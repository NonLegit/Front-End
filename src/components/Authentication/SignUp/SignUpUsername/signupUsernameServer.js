import { useEffect, useState } from 'react';
import { generateUsername } from 'unique-username-generator';

// mui components
import DoneIcon from '@mui/icons-material/Done';

// services
import axios from '../../../../services/instance';

// styles
import { wrongIcon, rightIcon } from '../../styles';
import theme from '../../../../styles/theme/index';

// Scripts
import { redditCookie } from '../../authenticationServer';
import { redirectHome } from '../../../../utils/Redirect';

// utils
import replaceDashWithUnderScore from '../../../../utils/replaceDashWithUnderScore';
/**
 *
 * check if username length [3-20], if valid username syntax,if username is unique and change state object accordingly
 * @param {string} username  --username to check for
 * @param {sstFunction} setUserName --setFunction for the userName State
 * @returns {void}
 */
export const checkUserNameSignUp = (username, setUserName) => {
  // console.log(username);// New value must be passsed
  // Check Username bwteen 3-20 characters
  if (username.length < 3 || username.length > 20) {
    setUserName((prevState) => ({
      ...prevState,
      color: theme.palette.error.main,
      icon: wrongIcon,
      error: 'Username must be between 3 and 20 characters',
    }));
    return;
  }

  // Check Invalid UserName
  if (!/^[A-Za-z0-9_-]*$/.test(username)) {
    // console.log('invalid');
    setUserName((prevState) => ({
      ...prevState,
      color: theme.palette.error.main,
      icon: wrongIcon,
      error: 'Letters, numbers, dashes, and underscores only. Please try again without symbols.',
    }));
    return;
  }

  // Check Unique Username
  axios.get(`/users/username_available?userName=${username}`).then((response) => {
    if (response.status === 200) {
      if (response.data.available === true) {
        // Unique
        setUserName((prevState) => ({
          ...prevState,
          color: theme.palette.primary.main,
          icon: rightIcon,
          error: null,
        }));
      } else {
        // Repeated
        setUserName((prevState) => ({
          ...prevState,
          color: theme.palette.error.main,
          icon: wrongIcon,
          error: 'That username is already taken.',
        }));
      }
    }
  }).catch((error) => {
    console.log(error);
  });
};

/**
 *
 * SignUp new User Function  if there is no a problem with the fields and verification the API is called
 * else not
 * Check for the Password Strength
 * Check if Verified to continue SignUp
 * @param {object} email --email object to signup with
 * @param {object} userName --userName object to signup with
 * @param {object} password --password object to signup with
 * @param {setFunction} setPassword --setfunction for the password
 * @param {state} verified --bool if verified by the Captcha
 * @param {setfunction} setLoading --setfunction for the Laoding state of the button
 * @param {setfunction} setButtonText --setfunction for the Button Text
 * @param {setfunction} setDisabled --setfunction for the setting Button Disabled or Enabled
 * @param {setfunction} setRedirectCaption --setfunction for the Caption in SignUp Page
 * @returns {void}
 */
export const signUp = (
  email,
  userName,
  setuserName,
  password,
  setPassword,
  verified,
  setLoading,
  setButtonText,
  setDisabled,
  setRedirectCaption,
  setCookies,
  removeCookie,
  popUp = false,
  handleClose = null,
) => {
  // console.log(email);
  // console.log(userName);// true value but case no change wrong value
  // console.log(password);// true value but case no change wrong value
  setLoading(true);

  // check if there is any errors
  if (userName.error != null || password.error != null) {
    setLoading(false);
    return;
  }

  // check if Empty (case he didn't make any change in the input field)
  if (userName.input === '' || password.input === '') {
    setLoading(false);
    return;
  }

  // Weak Password
  if (password.score < 2) {
    setPassword((prevState) => ({
      ...prevState,
      color: theme.palette.error.main,
      icon: wrongIcon,
      error: 'This Password isn\'t acceptable',
    }));
    setLoading(false);
    return;
  }

  // check if verified
  if (!verified) {
    setLoading(false);
    return;
  }
  // SignUpEndPoint
  axios.post('/users/signup', {
    // console.log(userName.input, password.input, email.input);
    userName: userName.input,
    email: email.input,
    password: password.input,
  }).then((response) => {
    setLoading(false);
    if (response.status === 201 || response.status === 200) {
      setButtonText(<DoneIcon />);
      setDisabled(true);
      setRedirectCaption(true);
      // Add Reddit Cookie

      if (popUp === false) {
        redirectHome(1000);
        redditCookie(setCookies, removeCookie);
      } else {
        // PopUp window
        setTimeout(() => {
          redditCookie(setCookies, removeCookie);
          handleClose();
        }, 1000);
      }
    } else {
      console.log('Error');
    }
  }).catch((error) => {
    if (error.response.data.errorType === 1) {
      setLoading(false);
      // weak password
      setPassword((prevState) => ({
        ...prevState,
        color: theme.palette.error.main,
        icon: wrongIcon,
        error: 'This Password isn\'t acceptable',
      }));
    } else if (error.response.data.errorType === 2) {
      setLoading(false);
      // repeated mail or username but for our logic it is only email
      // Repeated
      setuserName((prevState) => ({
        ...prevState,
        color: theme.palette.error.main,
        icon: wrongIcon,
        error: 'That user already exists.',
      }));
    }
    console.log(error);
  });
};

/**
 * This function works as a server for generating random usernames
 *
 * @function generateRandomUsernamesServer
 * @param {number} numberOfGeneratedUsernames - the maximum number of random usernames to be generated
 * @param {React.State} deps - the dependencies on which random usernames generated
 * @returns {Array.<string>} generated usernames
 */
export const generateRandomUsernamesServer = (numberOfGeneratedUsernames, deps) => {
  const [generatedUsernames, setGeneratedUsernames] = useState(null);
  useEffect(() => {
    (async () => {
      let counter = 0;
      const temp = [];
      do {
        let error = '';
        let newUserName;
        do {
          newUserName = generateUsername();
          newUserName = newUserName.slice(0, 20);
        }
        while (newUserName.length < 3);
        const username = replaceDashWithUnderScore(newUserName);
        const response = await axios.get('/users/username_available', {
          params: {
            userName: username,
          },
        }).catch((e) => {
          error = e;
        });
        const { data, status: statusCode } = response;
        if (error) {
          setGeneratedUsernames(['something went wrong']);
          return;
        }

        if (statusCode === 400) {
          const { message } = data;
          setGeneratedUsernames([message]);
          return;
        }

        if (data.available === true) {
          // unique
          counter += 1;
          temp.push(username);
        }
      } while (counter < numberOfGeneratedUsernames);
      // console.log(temp);
      setGeneratedUsernames(temp);
    })();
  }, [deps]);
  return generatedUsernames;
};

export default generateRandomUsernamesServer;
