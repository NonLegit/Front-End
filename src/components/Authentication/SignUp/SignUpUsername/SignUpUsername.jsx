import { useState } from 'react';

// mui components
import { IconButton, Typography } from '@mui/material';
import UpdateIcon from '@mui/icons-material/Update';

// Components
import PasswordStrengthBar from 'react-password-strength-bar';
import ReCAPTCHA from 'react-google-recaptcha';

// styles
import {
  BackLink, Body, ChooseUsernameContainer, Footer, Header, SuggestionBox, SuggestLink, TopBox,
} from './styles';
import {
  FirstPartyContainer, RedditTextField, rightIcon, RedditLoadingButton,
} from '../../styles';

// server

import { checkUserNameSignUp, signUp, generateRandomUsernamesServer } from './signupUsernameServer';

// scripts
import { checkPassword } from '../../authenticationServer';

import theme, { fonts } from '../../../../styles/theme';

// env Variables
const { REACT_APP_SITEKEY } = process.env;

/**
 * SignUp Username and Password Page View
 *
 * @component
 * @property {SetFunction} --setUserNamePage setView to Email Page on Back Button Click
 * @property {object} --email setView to Email Page on Back Button Click
 * @property {object} --userName userName to SignUp with
 * @property {function} --setUserName to set UserName
 * @property {object} --password password to SignUp with
 * @property {function} --setPassword to set password
 * @returns {React.Component} UserName and Password Form
 */
function SignUpUsername({
  setUserNamePage, email, userName, setUserName, password, setPassword, setCookies, removeCookie,
}) {
  // states
  const [defaultUserNameValue, setdefaultUserNameValue] = useState(userName?.input);
  const [defaultPasswordValue, setdefaultPasswordValue] = useState(password?.input);
  const [verified, setVerified] = useState(false);
  const [loading, setLoading] = useState(false);
  const [buttonTxt, setButtonText] = useState('Sign up');
  const [disabled, setDisabled] = useState(false);
  const [redirectCaption, setRedirectCaption] = useState(false);
  const [click, setClick] = useState(false);

  // server generate Random Names
  const generatedUsernames = generateRandomUsernamesServer(5, click);

  const signUpFunction = () => {
    checkUserNameSignUp(userName?.input, setUserName);
    checkPassword(password?.input, setPassword, password);
    signUp(email, userName, setUserName, password, setPassword, verified, setLoading, setButtonText, setDisabled, setRedirectCaption, setCookies, removeCookie);
  };

  return (
    <ChooseUsernameContainer data-testid="signup-username-test">
      <Header>
        <Typography variant="h1" fontSize={18} fontWeight={700} fontFamily="ibm-plex-sans,sans-serif" marginBottom="10px">
          Choose your username
        </Typography>
        <Typography fontSize={14}>
          Your username is how other community members will see you. This name will be used to
          {' '}
          credit you for things you share on Reddit. What should we call you?
        </Typography>
      </Header>
      <Body>
        <FirstPartyContainer width="350px" mnwidth="200px">
          <RedditTextField
            label="Choose a username"
            variant="filled"
            required
            InputProps={{
              endAdornment: (
                userName?.icon
              ),
              disableUnderline: true,
            }}
            clr={userName?.color}
            onChange={(e) => {
              setUserName((prevState) => ({
                ...prevState,
                input: e.target.value.trim(),
              }));
              checkUserNameSignUp(e.target.value.trim(), setUserName);
              setdefaultUserNameValue(e.target.value.trim());
            }}
            value={defaultUserNameValue || ''}
            data-testid="username-signup-field-test"
            helperText={userName?.error}
          />
          <PasswordStrengthBar
            password={password?.input}
            minLength={4}
            onChangeScore={(e) => (setPassword((prevState) => ({
              ...prevState,
              score: e,
            })))}
            scoreWords={[]}
            shortScoreWord=""
          />
          <RedditTextField
            label="password"
            variant="filled"
            type="password"
            required
            InputProps={{
              endAdornment: (
                password?.icon
              ),
              disableUnderline: true,
            }}
            clr={password?.color}
            onChange={(e) => {
              setPassword((prevState) => ({
                ...prevState,
                input: e.target.value.trim(),
              }));
              checkPassword(e.target.value.trim(), setPassword, undefined);
              setdefaultPasswordValue(e.target.value.trim());
            }}
            value={defaultPasswordValue || ''}
            helperText={password?.error}
            data-testid="password-signup-field-test"
          />
          <ReCAPTCHA
            sitekey={REACT_APP_SITEKEY}
            onExpired={() => setVerified(false)}
            onChange={() => setVerified(true)}
            size="normal"
            sx={{ width: '320px' }}
          />
        </FirstPartyContainer>
        <SuggestionBox>
          <TopBox>
            <Typography variant="p" fontSize={14}>
              Here are some username suggestions
            </Typography>
            <IconButton aria-label="delete" color="primary" onClick={() => setClick(!click)}>
              <UpdateIcon />
            </IconButton>
          </TopBox>
          {generatedUsernames ? (generatedUsernames.map((i) => (
            <SuggestLink
              key={i}
              onClick={() => {
                setUserName((prevState) => ({
                  ...prevState,
                  color: theme.palette.primary.main,
                  icon: rightIcon,
                  error: null,
                  input: i,
                }));
                setdefaultUserNameValue(i);
              }}
              href="#"
              helperText={password.error}
              data-testid={`suggest-username-${i}-test`}
            >
              {i}
            </SuggestLink>
          ))) : null}
        </SuggestionBox>
      </Body>
      {redirectCaption
        ? <Typography color="primary" fontSize="12px" fontFamily={fonts['system-ui']} fontWeight="600" margin="auto">You are now logged in. You will soon be redirected</Typography>
        : null}
      <Footer>
        <BackLink onClick={() => { if (disabled) return; setUserNamePage(false); }} href="#">Back</BackLink>
        <RedditLoadingButton
          width="155px"
          onClick={signUpFunction}
          loading={loading}
          disabled={disabled}
          data-testid="sigup-btn-test"
        >
          {buttonTxt}
        </RedditLoadingButton>
      </Footer>
    </ChooseUsernameContainer>
  );
}
export default SignUpUsername;
