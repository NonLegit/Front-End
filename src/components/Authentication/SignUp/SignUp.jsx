import { useEffect, useState } from 'react';
import { useCookies } from 'react-cookie';

// components
import SignUpEmail from './SignUpEmail/SignUpEmail';
import SignUpUsername from './SignUpUsername/SignUpUsername';
import LoadingPage from '../LoadingPage/LoadingPage';

// styles
import { AuthenticationConatiner, AuthenticationBG, AuthenticationBody } from '../styles';
import theme from '../../../styles/theme';

// scripts

import { redditCookie } from '../authenticationServer';

/**
 * SignUp Page Main Componnet with 2 view one for the Email and the Other for the userName
 *
 * @component
 * @returns {React.Component} SignUp Page
 */
function SignUp() {
  // states
  const [remeberMe, setremeberMe] = useState(false);
  const [userNamePage, setUserNamePage] = useState(false);
  const [email, setEmail] = useState({
    input: '', color: theme.palette.neutral.main, icon: null, error: null,
  });
  const [userName, setUserName] = useState({
    input: '', color: theme.palette.neutral.main, icon: null, error: null,
  });
  const [password, setPassword] = useState({
    input: '', color: theme.palette.neutral.main, icon: null, error: null, score: 0,
  });

  // cookies
  const [cookies, setCookies, removeCookie] = useCookies(['redditUser']);

  // useEffect
  useEffect(() => {
    // Check on Cookies
    // developememt
    /// Ask if this user already loggied in the back
    redditCookie(setCookies, removeCookie);
    // if logged in in the back end
    if (cookies.redditUser !== undefined) {
      // production
      // Update Cookie
      // redditCookie(setCookies);
      // check on Reddit cookie
      // if (cookies.redditUser === undefined) {
      //   redditCookie(setCookies);
      // }
      // Redirect to loading page
      console.log('Remember me is set to true');
      setremeberMe(true);
    } else {
      // No Cookie by Back End
      console.log('Remember me is set to false');
      setremeberMe(false);
    }
  }, []);

  const signUpView = userNamePage
    ? <SignUpUsername setUserNamePage={setUserNamePage} email={email} userName={userName} setUserName={setUserName} password={password} setPassword={setPassword} setCookies={setCookies} removeCookie={removeCookie} />
    : <SignUpEmail setUserNamePage={setUserNamePage} email={email} setEmail={setEmail} />;

  return (
    <AuthenticationConatiner data-testid="signup-test">
      {remeberMe
        ? (
          <>
            <AuthenticationBG />
            <AuthenticationBody mnwidth="280px" mxwidth="280px" data-testid="signup-email-test">
              <LoadingPage />
            </AuthenticationBody>
          </>
        )
        : signUpView}
    </AuthenticationConatiner>
  );
}
export default SignUp;
