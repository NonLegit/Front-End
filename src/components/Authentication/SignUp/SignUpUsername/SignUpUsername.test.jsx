import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import SignUp from '../SignUp';
import SignUpUsername from './SignUpUsername';

import replaceDashWithUnderScore from '../../../../utils/replaceDashWithUnderScore';

// Testing Generate user Name
describe(replaceDashWithUnderScore, () => {
  const testCases = [{
    input: 'aaa-aaaa-aa--aa',
    output: 'aaa_aaaa_aa__aa',
  },
  {
    input: 'basma',
    output: 'basma',
  },
  {
    input: 'Adha-m',
    output: 'Adha_m',
  },
  {
    input: '-',
    output: '_',
  },
  {
    input: '_',
    output: '_',
  },
  {
    input: '------',
    output: '______',
  },
  {
    input: '________',
    output: '________',
  },
  {
    input: 'basm-asds_dfg-',
    output: 'basm_asds_dfg_',
  },
  {
    input: '-_-_-_-_-_-_',
    output: '____________',
  },
  {
    input: 'Basma-Elhoseny',
    output: 'Basma_Elhoseny',
  },
  {
    input: 'teen-agecrick',
    output: 'teen_agecrick',
  },
  ];
  testCases.forEach((testCase, index) => {
    it(`Test Case #${index} : replaceDashWithUnderScore works correctly with input = ${testCase.input}, output = ${testCase.output}`, () => {
      expect(replaceDashWithUnderScore(testCase.input, testCase.output)).toBe(testCase.output);
    });
  });
});

// test snapshot
test('SigUp test snapshot', async () => {
  const tree = renderer.create(<SignUp><SignUpUsername /></SignUp>).toJSON();
  expect(tree).toMatchSnapshot();
});
