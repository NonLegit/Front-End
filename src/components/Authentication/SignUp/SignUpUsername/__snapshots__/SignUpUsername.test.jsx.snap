// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SigUp test snapshot 1`] = `
<div
  className="MuiBox-root css-pdterq"
  data-testid="signup-test"
>
  <div
    className="MuiBox-root css-1i4eqyw"
  />
  <div
    className="MuiBox-root css-rjktpo"
    data-testid="signup-email-test"
    mnwidth="280px"
    mxwidth="280px"
  >
    <div
      className="MuiBox-root css-eyuwdo"
    >
      <p
        className="MuiTypography-root MuiTypography-body1 css-1sg842v-MuiTypography-root"
      >
        Sign up
      </p>
      <p
        className="MuiTypography-root MuiTypography-body1 css-68wmct-MuiTypography-root"
      >
        By continuing, you are setting up a Reddit account and agree to our
         
        <a
          capital="none"
          className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-y1krgh-MuiTypography-root-MuiLink-root"
          href="https://www.redditinc.com/policies/user-agreement"
          onBlur={[Function]}
          onFocus={[Function]}
        >
          User Agreement
        </a>
         
        and
         
        <a
          capital="none"
          className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-y1krgh-MuiTypography-root-MuiLink-root"
          href="https://www.reddit.com/policies/privacy-policy"
          onBlur={[Function]}
          onFocus={[Function]}
        >
          Privacy Policy 
        </a>
        .
      </p>
    </div>
    <div
      className="MuiBox-root css-1vt6mgo"
    >
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium css-1inhtju-MuiButtonBase-root-MuiButton-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <img
          alt="Google"
          className="css-1x4dpap"
          src="google.png"
        />
        continue with google
      </button>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium css-1inhtju-MuiButtonBase-root-MuiButton-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <img
          alt="Facebook"
          className="css-1x4dpap"
          src="facebook.png"
        />
        continue with facebook
      </button>
    </div>
    <div
      className="MuiBox-root css-1rocah8"
    >
      <hr
        className="css-1tv3lka"
      />
      <p
        className="MuiTypography-root MuiTypography-body1 css-ikhzr9-MuiTypography-root"
      >
        OR
      </p>
      <hr
        className="css-1tv3lka"
      />
    </div>
    <form
      className="css-1notis6"
      data-testid="SignUpEmail-test"
      noValidate={true}
      onSubmit={[Function]}
      width="280px"
    >
      <div
        className="MuiFormControl-root MuiTextField-root css-fp4mq4-MuiFormControl-root-MuiTextField-root"
        clr="#978A8C"
      >
        <label
          className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-filled MuiFormLabel-colorPrimary Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-filled css-1mog9k0-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink={false}
          htmlFor=":r0:"
          id=":r0:-label"
        >
          Email
          <span
            aria-hidden={true}
            className="MuiFormLabel-asterisk MuiInputLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
          >
            â€‰
            *
          </span>
        </label>
        <div
          className="MuiInputBase-root MuiFilledInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-sizeSmall css-1pw81iq-MuiInputBase-root-MuiFilledInput-root"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            autoComplete="off"
            autoFocus={false}
            className="MuiInputBase-input MuiFilledInput-input MuiInputBase-inputSizeSmall css-c5v1qu-MuiInputBase-input-MuiFilledInput-input"
            disabled={false}
            id=":r0:"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            required={true}
            type="text"
            value=""
          />
        </div>
      </div>
      <button
        className="MuiButtonBase-root MuiButton-root MuiLoadingButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiLoadingButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-15fwilh-MuiButtonBase-root-MuiButton-root-MuiLoadingButton-root"
        data-testid="email-btn-test"
        disabled={false}
        id=":r1:"
        onBlur={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="submit"
      >
        continue
      </button>
    </form>
    <p
      className="MuiTypography-root MuiTypography-body1 MuiTypography-paragraph css-4owvyy-MuiTypography-root"
    >
      Already a redditor?
       
      <a
        capital="uppercase"
        className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1xn5q3i-MuiTypography-root-MuiLink-root"
        href="/login"
        onBlur={[Function]}
        onFocus={[Function]}
      >
        log in
      </a>
    </p>
  </div>
</div>
`;
